src/codec_impl.rs

```diff
@@ -0,0 +1,19 @@
+#[macro_export]
+/// implement parity codec for type
+macro_rules! impl_codec_for {
+    ($type:ty) => {
+        impl codec::Encode for $type {
+            fn encode(&self) -> Vec<u8> {
+                let encoded = bincode::serialize(&self).unwrap();
+                codec::Encode::encode(&encoded)
+            }
+        }
+
+        impl codec::Decode for $type {
+            fn decode<I: codec::Input>(value: &mut I) -> std::result::Result<Self, codec::Error> {
+                let decoded: Vec<u8> = codec::Decode::decode(value)?;
+                bincode::deserialize(decoded.as_slice()).map_err(|_| codec::Error)
+            }
+        }
+    };
+}
```

Comment:
That's probably not how the Parity SCALE codec is meant to encode things: using bincode inside.

Not sure if it matters, but maybe it would be more in line with the intended usage to manually implement `codec::{Encode, Decode}` only for the innermost types, and `#[derive(Encode, Decode)]` for the rest, similar to what we're doing with serde?

I'm also a bit confused about whether and to what extend `parity-scale-codec` supports serde itself; at least it has it as an optional dependency. If `serde` serializable types can just be directly serialized using SCALE anyway, there would be no need to explicitly support it in the first place?

Reply:
It seems `codec` cannot just derive for `serde` :(

Reply:
But upgrading pairing is kind of pain, maybe just keep it now